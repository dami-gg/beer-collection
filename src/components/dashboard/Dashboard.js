// @flow

import React, {Component} from 'react';
import DashboardOption from './dashboard-option/DashboardOption';

import './dashboard.scss';

import crate from '../../assets/images/crate.png';
import tap from '../../assets/images/tap.png';

class Dashboard extends Component {
  render() {
    return (
        <div className="dashboard">
          <DashboardOption
              title="Add beer to your collection"
              url="/beer/add"
              image={tap}
          />
          <DashboardOption
              title="See your collection"
              url="/collection"
              image={crate}
          />
        </div>
    );
  }
}

export default Dashboard;
