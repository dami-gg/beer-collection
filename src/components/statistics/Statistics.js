// @flow
import React, { PureComponent } from "react";

import Dashboard from "../dashboard/Dashboard";
import { PAGE_OPTIONS } from "./statistics.constants";

type Props = {};

export class Statistics extends PureComponent<Props> {
  render() {
    return (
      <div className="statistics">
        <Dashboard items={PAGE_OPTIONS} />
      </div>
    );
  }
}

export default Statistics;
