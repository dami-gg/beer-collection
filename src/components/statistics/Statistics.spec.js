import React from 'react';
import Statistics from './Statistics';
import Dashboard from '../dashboard/Dashboard';

import {StaticRouter as Router} from 'react-router-dom';

import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

describe('Statistics', () => {
  let component,
      element,
      mock;

  beforeEach(mocks);

  it('should render without crashing', () => {
    shallow(<Statistics />);
  });

  it('should match the snapshot', () => {
    component = renderer.create(
        <Router context={mock.context}>
          <Statistics/>
        </Router>
    );
    let json = component.toJSON();

    expect(json).toMatchSnapshot();
  });

  describe('render', () => {
    beforeEach(setup);

    it('should render a statistics element', () => {
      element = component.find('.statistics');

      expect(element.length).toEqual(1);
    });

    it('should render a Dashboard component', () => {
      element = component.find(Dashboard);

      expect(element.length).toEqual(1);
    });
  });

  function setup() {
    render();
  }

  function render() {
    component = shallow(<Statistics />);
  }

  function mocks() {
    mock = {
      context: {}
    };
  }
});

