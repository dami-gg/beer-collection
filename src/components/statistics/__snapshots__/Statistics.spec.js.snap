// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statistics should match the snapshot 1`] = `
<div
  className="statistics">
  <div
    className="dashboard">
    <div
      className="dashboard__option">
      <a
        className="dashboard__option__card"
        href="/origin"
        onClick={[Function anonymous]}>
        <div
          className="dashboard__option__image-wrapper">
          <img
            alt="logo"
            className="dashboard__option__image"
            src="https://unsplash.it/150/150" />
        </div>
        <div
          className="dashboard__option__title-wrapper">
          <p
            className="dashboard__option__title">
            Origin
          </p>
        </div>
      </a>
    </div>
  </div>
</div>
`;
